homeassistant/components/go2rtc/server.py

```diff
@@ -70,6 +72,7 @@ def __init__(
         """Initialize the server."""
         self._hass = hass
         self._binary = binary
+        self._log_buffer: deque[str] = deque(maxlen=_LOG_BUFFER_SIZE)
```

Comment:
Do we need a deque if we're only adding to one side and clearing the queue? A list may be as fast for those operations?

Reply:
There are no lists with limited size, the deque is used for that purpose.

Reply:
Instead of buffering, should we always log the output of go2rtc with the appropriate level?
Every line includes the level. We only need to parse it. So at least we would have all entries

Reply:
I don't understand? If there's no error, the appropriate level is DEBUG, unless we can see the future?

Reply:
For example, this morning, I had an issue with one camera and go2rtc. All other camera streams were working normally. So, in general, go2rtc was working and did not crash, but an error was logged for the particular camera (some ffmpeg error). The user will not have this error in the logs, as go2rtc is still running.

Reply:
For context: @edenhaus means we should check the output of the go2rtc process to detect the severity of the logs, and then log in Home Assistant with a matching log level. I agree this makes perfect sense, I'll take care of it in a follow-up 👍